Storage/DoctrineStorage.php

```diff
@@ -150,7 +152,7 @@ private function tableExists() {
 	private function createTable() {
 		// TODO remove as soon as Doctrine DBAL >= 3.0 is required
 		$stringType = defined('Doctrine\DBAL\Types\Types::STRING') ? Types::STRING : Type::STRING;
-		$arrayType = defined('Doctrine\DBAL\Types\Types::ARRAY') ? Types::ARRAY : Type::TARRAY;
+		$arrayType = defined('Doctrine\DBAL\Types\Types::JSON') ? Types::JSON : Type::TARRAY;
```

Comment:
I think more work is needed if this change is included.

On `set` and `get` methods, I think we should detect the defined column type, get the `Doctrine\DBAL\Types\Type\Type` and then call `Type::convertToDatabaseValue()` and `Type::convertToPHPValue()` to set and get the right value.

Reply:
This change was needed to avoid triggering a deprecation notice. Do you spot an issue with it?

I didn't plan on extending the functionality. At least not right now as part of this PR. It's large enough already. 😄

Reply:
I was trying in https://github.com/franmomu/CraueFormFlowBundle/commit/fe4383f6089836db83ed3b03b7dfd21ad87f37cc resulting in https://github.com/franmomu/CraueFormFlowBundle/actions/runs/5997192724/job/16263113060

I guess sqlite manages `json` as `string`, but using MySQL or PostgreSQL looks like it would fail.

Yeah, so maybe it's better to ignore the deprecation for now and in another PR change that.

Reply:
Oh wow. Interestingly, I don't get these errors when testing locally, but indeed the tests should actually be run against MySQL as well.

As you suggest, I'll revert this change for now.